src/lib/components/tablev2/Tablestyle.js

```diff
@@ -26,7 +26,10 @@ export const TableHeader = styled.div`
 
 export const HeadRow = styled.div`
   height: 2.286rem;
-  width: 100% !important;
+  width: ${(props) =>
+    props.hasScrollBar
+      ? `calc(100% - ${props.scrollBarWidth}px - 4px )!important;` // -4px for border
```

Comment:
Ideally, we should extract the 4px to a prop.
If we hardcode the number here, one day when we update the border we are going to easily forget it here.